src/reactviews/pages/QueryResult/queryResultPane.tsx

```diff
@@ -269,6 +278,32 @@ export const QueryResultPane = () => {
         return grids;
     };
 
+    useEffect(() => {
+        const getExecutionPlanGraphs = () => {
+            if (
+                metadata &&
+                metadata.executionPlanState.xmlPlans &&
+                metadata.executionPlanState?.executionPlanGraphs &&
+                !metadata.executionPlanState.executionPlanGraphs.length
+            ) {
+                state!.provider.getExecutionPlan(
+                    metadata.executionPlanState.xmlPlans,
+                );
+            }
+        };
+
+        if (
+            metadata &&
+            metadata.resultSetSummaries &&
+            metadata.executionPlanState.xmlPlans &&
+            Object.keys(metadata.resultSetSummaries).length > 0 &&
+            Object.keys(metadata.resultSetSummaries).length ===
+                metadata.executionPlanState.xmlPlans.length
```

Comment:
Can you add some comments to this long boolean statements to say what each is meaning to check since the prop drilling doesn't match up (and therefore isn't immediately intuitive)?